obstore/python/obstore/auth/earthdata.py

```diff
@@ -6,157 +6,419 @@
 from __future__ import annotations
 
 import json
+import os
 from datetime import datetime
 from typing import TYPE_CHECKING
+from urllib.parse import urlparse
+
+from obstore.auth._http import default_aiohttp_session, default_requests_session
 
 if TYPE_CHECKING:
+    from collections.abc import Coroutine, Mapping
+    from typing import Any, Callable
+
+    import aiohttp
+    import aiohttp_retry
+    import requests
+
     from obstore.store import S3Config, S3Credential
 
-CREDENTIALS_API = "https://archive.podaac.earthdata.nasa.gov/s3credentials"
+
+EARTHDATA_HOST_OPS = "urs.earthdata.nasa.gov"
+EARTHDATA_HOST_UAT = "uat.urs.earthdata.nasa.gov"
```

Comment:
Can we have docstrings for each of these constants, which will get rendered in the docs?

We'll also need to add these constants to the docs here: https://github.com/developmentseed/obstore/blob/6d878a2e39e5bc26f87ce8505fac821b42f2689e/docs/api/auth/earthdata.md?plain=1#L1-L8

Reply:
Sure thing. I'm not familiar with the required syntax in the md file. Should I simply add 2 more entries similar to those 2 existing entries in the md file?